### ISSUE TITLE
Ed25519 signature functions: empty message is valid

### ISSUE THREAD

**Issue by philikon** [paixaop/node-sodium#77]
`crypto_sign_ed25519_detached(...)` and `crypto_verify_ed25519_detached(...)` should accept empty buffers for the `message` argument. Empty messages are perfectly valid, they work with libsodium, and the first test vector in the RFC Draft is in fact the empty message: https://tools.ietf.org/html/draft-josefsson-eddsa-ed25519-02#section-6

This is what happens when an empty buffer is supplied:
```
> var s = require('sodium')
> var k = s.api.crypto_sign_ed25519_keypair()
> s.api.crypto_sign_ed25519_detached(new Buffer(0), k.secretKey)
Error: argument message length cannot be zero
```

**Comment by paixaop** [comment]
Fixed in 1.1.6, please test it out

**Comment by philikon** [comment]
Thanks, will do!
